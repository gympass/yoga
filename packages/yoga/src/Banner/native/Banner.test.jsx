import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import { CheckedFull } from '@gympass/yoga-icons';
import { ThemeProvider, Banner } from '../..';

describe('<Banner />', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Banner message="Feedback message." />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with custom variant', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Banner variant="success" message="Success banner without button" />
        <Banner
          variant="informative"
          message="Informative banner without button"
        />
        <Banner variant="attention" message="Attention banner without button" />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with icon', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Banner icon={CheckedFull} message="Banner with icon" />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with button', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Banner
          message="Banner with button"
          primaryButton={{
            label: 'Action',
            action: () => {},
          }}
        />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should call button action function when the banner button is pressed', () => {
    const onButtonPressMock = jest.fn();
    const { getByText } = render(
      <ThemeProvider>
        <Banner
          message="Banner with button"
          primaryButton={{
            label: 'Action',
            action: onButtonPressMock,
          }}
        />
      </ThemeProvider>,
    );

    fireEvent.press(getByText('Action'));

    expect(onButtonPressMock).toHaveBeenCalled();
  });

  it('should match snapshot with two action buttons', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <Banner
          message="Banner with two action buttons"
          primaryButton={{
            label: 'Primary Action',
            action: () => {},
          }}
          secondaryButton={{
            label: 'Secondary Action',
            action: () => {},
          }}
        />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should call both buttons action function when each banner action button are pressed', () => {
    const onPrimaryButtonPressMock = jest.fn();
    const onSecondaryButtonPressMock = jest.fn();

    const { getByText } = render(
      <ThemeProvider>
        <Banner
          message="Banner with button"
          primaryButton={{
            label: 'Primary Action',
            action: onPrimaryButtonPressMock,
          }}
          secondaryButton={{
            label: 'Secondary Action',
            action: onSecondaryButtonPressMock,
          }}
        />
      </ThemeProvider>,
    );

    fireEvent.press(getByText('Primary Action'));
    expect(onPrimaryButtonPressMock).toHaveBeenCalled();

    fireEvent.press(getByText('Secondary Action'));
    expect(onSecondaryButtonPressMock).toHaveBeenCalled();
  });
});
