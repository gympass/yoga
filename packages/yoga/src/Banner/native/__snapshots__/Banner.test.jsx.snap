// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#E0DFFF",
        "borderRadius": 8,
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
    ]
  }
  variant="informative"
>
  <View
    alignItems="center"
    flex={1}
    flexDirection="row"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Text
      flex={1}
      marginVertical="xxsmall"
      style={
        Array [
          Object {
            "color": "#231B22",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "Rubik",
            "fontSize": 14,
            "fontWeight": "400",
            "lineHeight": 20,
            "marginBottom": 8,
            "marginTop": 8,
          },
        ]
      }
    >
      Feedback message.
    </Text>
  </View>
</View>
`;

exports[`<Banner /> should match snapshot with button 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#E0DFFF",
        "borderRadius": 8,
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
    ]
  }
  variant="informative"
>
  <View
    alignItems="center"
    flex={1}
    flexDirection="row"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Text
      flex={1}
      marginVertical="xxsmall"
      style={
        Array [
          Object {
            "color": "#231B22",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "Rubik",
            "fontSize": 14,
            "fontWeight": "400",
            "lineHeight": 20,
            "marginBottom": 8,
            "marginTop": 8,
          },
        ]
      }
    >
      Banner with button
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        marginLeft="xxsmall"
        pressed={false}
        small={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 9999,
              "flexDirection": "row",
              "height": 32,
              "justifyContent": "center",
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {
              "marginLeft": 8,
            },
          ]
        }
      >
        <Text
          color="#231B22"
          pressed={false}
          secondary={true}
          small={true}
          style={
            Array [
              Object {
                "color": "#231B22",
                "fontFamily": "Rubik",
                "fontSize": 14,
                "fontWeight": "500",
                "textAlign": "center",
              },
            ]
          }
        >
          Action
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<Banner /> should match snapshot with custom variant 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#C1EEDB",
          "borderRadius": 8,
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        },
      ]
    }
    variant="success"
  >
    <View
      alignItems="center"
      flex={1}
      flexDirection="row"
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Text
        flex={1}
        marginVertical="xxsmall"
        style={
          Array [
            Object {
              "color": "#231B22",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "Rubik",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "marginBottom": 8,
              "marginTop": 8,
            },
          ]
        }
      >
        Success banner without button
      </Text>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0DFFF",
          "borderRadius": 8,
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        },
      ]
    }
    variant="informative"
  >
    <View
      alignItems="center"
      flex={1}
      flexDirection="row"
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Text
        flex={1}
        marginVertical="xxsmall"
        style={
          Array [
            Object {
              "color": "#231B22",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "Rubik",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "marginBottom": 8,
              "marginTop": 8,
            },
          ]
        }
      >
        Informative banner without button
      </Text>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FCD6C3",
          "borderRadius": 8,
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        },
      ]
    }
    variant="attention"
  >
    <View
      alignItems="center"
      flex={1}
      flexDirection="row"
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Text
        flex={1}
        marginVertical="xxsmall"
        style={
          Array [
            Object {
              "color": "#231B22",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "Rubik",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
              "marginBottom": 8,
              "marginTop": 8,
            },
          ]
        }
      >
        Attention banner without button
      </Text>
    </View>
  </View>,
]
`;

exports[`<Banner /> should match snapshot with icon 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#E0DFFF",
        "borderRadius": 8,
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
    ]
  }
  variant="informative"
>
  <View
    alignItems="center"
    flex={1}
    flexDirection="row"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <RNSVGSvgView
      bbHeight={20}
      bbWidth={20}
      fill="#231B22"
      focusable={false}
      height={20}
      marginRight="xxsmall"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Array [
            Object {
              "height": 20,
              "marginRight": 8,
              "width": 20,
            },
          ],
          Object {
            "opacity": 1,
          },
          Object {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      width={20}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4280490786,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      />
    </RNSVGSvgView>
    <Text
      flex={1}
      marginVertical="xxsmall"
      style={
        Array [
          Object {
            "color": "#231B22",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "Rubik",
            "fontSize": 14,
            "fontWeight": "400",
            "lineHeight": 20,
            "marginBottom": 8,
            "marginTop": 8,
          },
        ]
      }
    >
      Banner with icon
    </Text>
  </View>
</View>
`;

exports[`<Banner /> should match snapshot with two action buttons 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#E0DFFF",
        "borderRadius": 8,
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
    ]
  }
  variant="informative"
>
  <View
    alignItems="center"
    flex={1}
    flexDirection="row"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Text
      flex={1}
      marginVertical="xxsmall"
      style={
        Array [
          Object {
            "color": "#231B22",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "Rubik",
            "fontSize": 14,
            "fontWeight": "400",
            "lineHeight": 20,
            "marginBottom": 8,
            "marginTop": 8,
          },
        ]
      }
    >
      Banner with two action buttons
    </Text>
  </View>
  <View
    display="flex"
    flex={1}
    flexDirection="row"
    justifyContent="flex-end"
    marginBottom="xxsmall"
    style={
      Array [
        Object {
          "display": "flex",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-end",
          "marginBottom": 8,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        marginRight="xxxsmall"
        pressed={false}
        small={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 9999,
              "flexDirection": "row",
              "height": 32,
              "justifyContent": "center",
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {
              "marginRight": 4,
            },
          ]
        }
      >
        <Text
          color="#231B22"
          pressed={false}
          secondary={true}
          small={true}
          style={
            Array [
              Object {
                "color": "#231B22",
                "fontFamily": "Rubik",
                "fontSize": 14,
                "fontWeight": "500",
                "textAlign": "center",
              },
            ]
          }
        >
          Primary Action
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pressed={false}
        small={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 9999,
              "flexDirection": "row",
              "height": 32,
              "justifyContent": "center",
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {},
          ]
        }
      >
        <Text
          color="#231B22"
          pressed={false}
          secondary={true}
          small={true}
          style={
            Array [
              Object {
                "color": "#231B22",
                "fontFamily": "Rubik",
                "fontSize": 14,
                "fontWeight": "500",
                "textAlign": "center",
              },
            ]
          }
        >
          Secondary Action
        </Text>
      </View>
    </View>
  </View>
</View>
`;
